discord/experiment.py

```diff
@@ -0,0 +1,733 @@
+
+        .. describe:: iter(x)
+
+            Returns an iterator of the resources eligible for the override.
+
+    .. versionadded:: 2.1
+
+    Attributes
+    -----------
+    experiment: :class:`GuildExperiment`
+        The experiment this override belongs to.
+    bucket: :class:`int`
+        The bucket the override applies.
+    ids: List[:class:`int`]
+        The eligible guild/user IDs for the override.
+    """
+
+    __slots__ = ('experiment', 'bucket', 'ids')
+
+    def __init__(self, experiment: GuildExperiment, data: OverridePayload):
+        self.experiment = experiment
+        self.bucket: int = data['b']
+        self.ids: Tuple[int, ...] = tuple(data['k'])
+
+    def __repr__(self) -> str:
+        return f'<ExperimentOverride bucket={self.bucket} ids={self.ids!r}>'
+
+    def __len__(self) -> int:
+        return len(self.ids)
+
+    def __contains__(self, item: Union[int, Snowflake]) -> bool:
+        return getattr(item, 'id', item) in self.ids
+
+    def __iter__(self) -> Iterator[int]:
+        return iter(self.ids)
+
+
+class HoldoutExperiment:
+    """Represents an experiment dependency.
+
+    .. container:: operations
+
+        .. describe:: x in y
+
+            Checks if a guild fulfills the dependency.
+
+    .. versionadded:: 2.1
+
+    Attributes
+    -----------
+    dependent: :class:`GuildExperiment`
+        The experiment that depends on this experiment.
+    name: :class:`str`
+        The name of the dependency.
+    bucket: :class:`int`
+        The required bucket of the dependency.
+    """
+
+    __slots__ = ('dependent', 'name', 'bucket')
+
+    def __init__(self, dependent: GuildExperiment, name: str, bucket: int):
+        self.dependent = dependent
+        self.name: str = name
+        self.bucket: int = bucket
+
+    def __repr__(self) -> str:
+        return f'<HoldoutExperiment dependent={self.dependent!r} name={self.name!r} bucket={self.bucket}>'
+
+    def __contains__(self, item: Guild) -> bool:
+        return self.is_eligible(item)
+
+    @property
+    def experiment(self) -> Optional[GuildExperiment]:
+        """Optional[:class:`GuildExperiment`]: The experiment dependency, if found."""
+        experiment_hash = murmurhash32(self.name, signed=False)
+        experiment = self.dependent._state.guild_experiments.get(experiment_hash)
+        if experiment and not experiment.name:
+            # Backfill the name
+            experiment._name = self.name
+        return experiment
+
+    def is_eligible(self, guild: Guild, /) -> bool:
+        """Checks whether the guild fulfills the dependency.
+
+        Parameters
+        -----------
+        guild: :class:`Guild`
+            The guild to check.
+
+        Returns
+        --------
+        :class:`bool`
+            Whether the guild fulfills the dependency.
+        """
+        experiment = self.experiment
+        if experiment is None:
+            # We don't have the experiment, so we can't check
+            return True
```

Comment:
Shouldn’t this be False or just raise an error?

Reply:
I don't think it's worth erroring, feels like it'd be pretty common to not have the experiment in-cache (e.g. when fetching)